spyder/app/mainwindow.py

```diff
@@ -799,6 +799,7 @@ def create_edit_action(text, tr_text, icon):
         from spyder.plugins.editor import Editor
         self.editor = Editor(self)
         self.editor.register_plugin()
+        self.editor.dockwidget.topLevelChanged.connect(self.undock_editor)

```

Comment:
This line can also be moved to the Editor plugin. We can make it the last one of its `register_plugin` method.